File: src/models/review.model.ts
// src/models/review.model.ts
export interface Review {
    id?: number;
    place_id: number;
    user_id: number;
    rating: number;
    comment?: string;
    created_at?: Date;
    created_by_username?: string; // Для отображения имени автора отзыва
}

export interface CreateReviewDto {
    rating: number;
    comment?: string;
}

File: src/models/review_model.dart
// lib/models/review_model.dart
class Review {
  final int id;
  final int rating;
  final String? comment;
  final DateTime createdAt;
  final String? createdByUsername;

  Review({
    required this.id,
    required this.rating,
    this.comment,
    required this.createdAt,
    this.createdByUsername,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: json['id'],
      rating: json['rating'],
      comment: json['comment'],
      createdAt: DateTime.parse(json['created_at']),
      createdByUsername: json['created_by_username'],
    );
  }
}